litestar/contrib/jinja.py

```diff
@@ -29,15 +29,26 @@
 class JinjaTemplateEngine(TemplateEngineProtocol["JinjaTemplate"]):
     """The engine instance."""
 
-    def __init__(self, directory: DirectoryPath | list[DirectoryPath]) -> None:
+    def __init__(
+        self,
+        directory: DirectoryPath | list[DirectoryPath] | None = None,
+        engine_instance: Environment | None = None,
```

Comment:
Is this `engine_instance` or `environment`? Let's find the best semantics

Reply:
I settled for `engine_instance` over `environment` because it calls the `TemplateEngineProtocol` `__init__` and the latter deals with mako as well, and since mako doenst have an `Environment` but a `TemplateLookup` I found using `environment` was better